include/amici/model.h

```diff
@@ -504,6 +504,32 @@ namespace amici {
          */
         int checkFinite(const int N,const realtype *array, const char* fun) const;
 
+
+        /**
+         * @brief Get names of the model parameters
+         * @return the names
+         */
+        virtual std::vector<std::string> getParameterNames() const { return std::vector<std::string>(); }
```

Comment:
can we have smarter defaults here? Maybe something like `unnamedParameter1,unnamedParameter2,...`

Reply:
Technically yes. I thought this way it is easier to check if names are set or not. Creating default names by the user is trivial. I find it unlikely that the user really wants to use any of the default names we could come up with here.

Reply:
Fair enough, but I think we should then at least print a warning/error message or even throw an exception.

Reply:
Hmm... I think I'd prefer: 
```
auto names = model.getParameterNames();
if(names.empty()) {
// ... 
}
```
over
```
try {
auto names = model.getParameterNames();
} catch (AmiException &e) {
// ... 
}
```

Reply:
You actually need to do:
```
auto names = model.getParameterNames();
if(names.empty() && model.getNumberOfParameters()>0) {
// ... 
}
```
The fact that the result for zero length parameters and no set Names is the same bothers me. Maybe introduce a function `isSetParameterNames()` that checks `if(names.empty() && model.getNumberOfParameters()>0)`?

Reply:
Okay. I think I'd prefer that. I think `hasParameterNames()` would also be coherent with libsbml. We can also reopen that discussion after having used it in practice a bit more.

Reply:
sounds good!